source/input.ts

```diff
@@ -26,17 +27,54 @@ export function createInputManager(): InputManager {
    * Filter and apply listeners that matches the event's properties.
    *
    * @param {any} chunk
-   * @param {readline.key} key - The eveny key.
+   * @param {readline.key} key - The event key.
    */
   function onKeyPress(chunk: any, key: readline.Key): void {
     listeners
       .filter(
         filterInputListener, key
-      ).map(
-        listener => listener.event()
-      );
+      ).map(executeEvent);
   }
 
+  function isPromise(obj: any) {
+    return Boolean(obj.then)
+  }
+
+  function executeEvent(listener: InputListener) {
+    unbindKeypress();
+    const eventResult = listener.event()
+
+    if (eventResult && isPromise(eventResult)) {
+      eventResult.then(data => {
+        if (data && data.usingInquirer) {
+          reopenStdinAfterInquirer();
```

Comment:
`inquirer` was preventing `keypress` event to work 😢 
So, this method will fix it

Ref: https://github.com/SBoudrias/Inquirer.js/issues/662